indexer.go

```diff
@@ -169,15 +156,57 @@ func (i *Indexer) IndexBlock(blockHeight int) error {
 		return ErrBlockHasNoHash
 	}
 
-	err = i.writer.WriteBlock(convertProviderBlockToBlock(blockOutput))
+	blockTxs, err := getProviderBlockTransactions(blockHeight, i.provider)
+
+	if err != nil {
+		return err
+	}
+
+	relayCount := 0
+
+	for _, tx := range blockTxs {
+		if tx.TxResult.MessageType == "claim" {
+			proofs := tx.StdTx.Msg.Value["total_proofs"].(int)
+			relayCount += proofs
+		}
+	}
+
+	block := convertProviderBlockToBlock(blockOutput, relayCount)
+
+	err = i.writer.WriteBlock(block)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func convertProviderBlockToBlock(providerBlock *provider.GetBlockOutput) *Block {
+func getProviderBlockTransactions(blockHeight int, indexerProvider Provider) ([]*provider.Transaction, error) {
```

Comment:
This function should be a method of `Indexer`, in that way you don't need to pass provider as a param